### ISSUE TITLE
New request: The Terraria Wiki

### ISSUE THREAD

**Issue by PoggChamp** [openzim/zim-requests#735]
- Website URL: https://terraria.wiki.gg/
- License: **Creative Commons**
- Desired ZIM Title: Terraria Wiki
- Desired ZIM Description: the comprehensive reference written and maintained by the players.
- Desired ZIM Icon –png (URL or attach one): https://terraria.wiki.gg/favicon.ico
- Language (ISO 639-3): **eng**
- Is this a MediaWiki?: **yes**

**Comment by RavanJAltaie** [comment]
The recipe created
https://farm.openzim.org/recipes/terrariawiki_en
Will update the library link once ready

**Comment by RavanJAltaie** [comment]
Troubleshooting the recipe https://github.com/openzim/mwoffliner/issues/1948

**Comment by outtieTV** [comment]
I'd like to see a Terraria wiki zim file since Terraria had a final update. Would be useful to put on my iiab.

**Comment by Popolechien** [comment]
@outtieTV Site seems to be down/moving, and from the mwoffliner issue above it looks like it has a Cloudflare in place that will prevent us from crawling. Any chance you can reach out to them (cc hello@kiwix.org) and ask them to whitelist us?

**Comment by benoit74** [comment]
Last attempt on Zimfarm was a bit old, I've just requested it again to give it a fresh try

**Comment by benoit74** [comment]
I've just requested a new task with latest mwoffliner, we've again made a significant amount of changes : https://farm.openzim.org/pipeline/34c933fd-ddb5-4efc-a134-5145f0893d92

And a bug was reported at [Broken page layout in Terraria wiki (wiki.gg) ZIM at large widths (>900px, >=1120px) mwoffliner#2562](https://github.com/openzim/mwoffliner/issues/2562)

**Comment by D-side** [comment]
I'll have a proper long stroll around the new file (2025-10) in the coming days, but I've had a quick look just now.
The design suffered a bit (its nice background art is always obscured now), but the content I've quickly looked at is now displayed reliably. Some spacing from screen edges might be nice, but that's very much a cosmetic thing.

Thanks @benoit74!

**Comment by D-side** [comment]
Spotted a problem in the latest file, math doesn't render, such as formulas on [page regarding Terraria's luck mechanic](https://terraria.wiki.gg/wiki/Luck).

The TeX markup is still there as well as `[math]` and `[/math]` delimiters, but they appear in the content as plain unrendered source text in muted colors.

I'm a pretty casual player so I don't find this very important, but especially dedicated players might be expecting to see rendered formulas.

**Comment by kelson42** [comment]
Indeed, see https://dev.library.kiwix.org/viewer#terrariawiki_en_maxi_2025-10/Luck

@benoit74 Are we impacted here by https://github.com/openzim/mwoffliner/issues/1371 ?

**Comment by D-side** [comment]
Another difference I've noticed is a lack of sounds, e. g. see the info card at the top for Meowmere [in the original](https://terraria.wiki.gg/wiki/Meowmere) and [in the new file](https://dev.library.kiwix.org/viewer#terrariawiki_en_maxi_2025-10/Meowmere), the section titled "Sounds".

edit: music, with a different kind of player, is missing too: [original](https://terraria.wiki.gg/wiki/Music#In-game_tracks), [zim](https://dev.library.kiwix.org/viewer#terrariawiki_en_maxi_2025-10/Music#In-game_tracks).

Also noticed that table of contents is missing (in the music page at least), though heading anchors seem to work.

**Comment by benoit74** [comment]
Regarding math formula, we are indeed impacted by https://github.com/openzim/mwoffliner/issues/1371 ; hopefully will be fixed in somewhere in 2026.

Regarding absence of sound, it is normal, the Zimfarm recipe requested a `novid:maxi` flavor, and `novid` means no video and no sound. Is sound very important or a nice to have? It will probably significantly increase the ZIM size. Should we have two ZIMs, one with sound and one without?

Regarding the table of content, this is normal as well, table of contents are to be handled by the ZIM reader, and not all ZIM readers have implemented it equally well.

**Comment by D-side** [comment]
Ohh, right, for a `novid` flavor that makes sense.

I'm not sure the file size would be **that** much bigger with sound included, the game's total size is under 700MB and I don't think it employs any asset generation, though different formats I suppose could cause an increase beyond that, in theory. The game also reuses many of its SFX for many purposes and from what I can tell, the reuse is reflected on the wiki (pages using the same effect point at the same sound file). Most of the sounds (almost all except the music) are also very short. Sound cues in the game are very optional, but helpful, so those would be nice to have.

And the current ZIM file already has a number of very video-like GIF animations such as 3.45Mb demonstration of the [Scarab Bomb](https://dev.library.kiwix.org/viewer#terrariawiki_en_maxi_2025-10/Scarab_Bomb) (a lower floating block on the right). I'd guesstimate that those would still be the top contributor to the file size even with sounds. In fact, there might be substantial compression potential in reencoding those from GIF to video. I tried a plain `ffmpeg -i $in.gif $out.mp4` without any specific options and it squished those 3.45MB to just 1.05MB (using H264) and I'm not noticing any loss in quality.

I'd try building a ZIM file with sounds for comparison. Maybe there's not much point in the `novid` flavor in the first place.

**Comment by benoit74** [comment]
I've just requested a ZIM with sounds and videos: https://farm.openzim.org/pipeline/547bb059-bcba-4b44-974c-4d1d8e2f4564

We propose with @Popolechien to publish the ZIM despite Mathjax issue, it is not a blocker point from our PoV. 

Let's see if ZIM with sound has an appropriate size, if yes we will publish this one, otherwise I would suggest to create both `novid` and `all` ("with vid") ZIMs..

**Comment by D-side** [comment]
The size difference ended up being [1.26GiB `maxi` (I assume it's `novid`) vs. 1.40GiB for the whole thing](https://dev.library.kiwix.org/#lang=eng&q=Terraria), which is probably negligible. And [sounds work fine](https://dev.library.kiwix.org/viewer#terrariawiki_en_2025-10/Meowmere).

I also don't see MathJAX as critical for that particular wiki. I take it it's a bigger issue in other wikis so there are hopes of fixing this eventually and that the fix would eventually reach Terraria wiki as well. That sounds fine by me.

An additional bit of context on future updates: since the game is being updated to this day (with a new 1.4.5 update currently in the works), periodic updates would be helpful. Not until the next update probably, but certainly for some time after it releases. If feasible, of course.

**Comment by benoit74** [comment]
@D-side thank you

We aim to update the ZIM quarterly. It is fully automated. It allows to reasonably grab any edits made in the Wiki, and to benefit from any new feature / bug fix done in the scraper.

You can always request (with a new issue in this Github Repo) a manual run between these quarterly updates if for some reason the ZIM is too outdated because many edits have been done in the wiki (typically following a major release). But we prefer when we can avoid them to save our efforts for other stuff ^^

**Comment by D-side** [comment]
@benoit74 understood, thanks! All set for publishing the full ZIM file then?

---

Also, regarding the idea I mentioned above about converting `*.gif`s to `*.mp4`s for some space savings, I take it [mwoffliner](https://github.com/openzim/mwoffliner/) would be the right place for it?

**Comment by benoit74** [comment]
Yes, mwoffliner is the place to implement it. You can open an issue there to discuss the matter first, not sure it is really that straightforward

**Comment by GitEin11** [comment]
on stats with master/expert e.g. monsters pages clicking the button (master/expert) doesnt update the content

**Comment by benoit74** [comment]
Do you have a URL to share, or the precise title of the article at least?

I strongly suspect this is a Javascript "thing", and Javascript is not supported ATM (see https://github.com/openzim/mwoffliner/pull/2483)

**Comment by GitEin11** [comment]
> Do you have a URL to share, or the precise title of the article at least?
> 
> I strongly suspect this is a Javascript "thing", and Javascript is not supported ATM (see [openzim/mwoffliner#2483](https://github.com/openzim/mwoffliner/pull/2483))

any monster page, well I guess we'll gonna wait till JavaScript is supported.

https://github.com/user-attachments/assets/aa0d9131-9601-4ed4-9486-4b2b87c8dc1f

**Comment by D-side** [comment]
Yep.

Example: [original](https://terraria.wiki.gg/wiki/Queen_Slime), [latest file](https://dev.library.kiwix.org/viewer#terrariawiki_en_2025-10/Queen_Slime) (click the colored Classic/Expert/Master inside the info card on the right)

**Comment by benoit74** [comment]
Yep, this is JS code... not sure yet where it comes from, but you will have to live without it indeed.

**Comment by D-side** [comment]
It's probably a bigger issue compared to sound or math, as "Expert mode" has a number of exclusive things, which makes some of the tabs' contents very different, particularly monster item drops. And someone who might want an offline version of Terraria Wiki would probably be interested in the game enough to play on Expert or Master.

So I'd suggest postponing proper publication until this is sorted out.

**Comment by Popolechien** [comment]
@D-side @GitEin11 Do you reckon we nevertheless have enough to move this to PROD, or would you rather wait until the above issues are fixed (meaning that for the time being and without these the ZIM is nice but not super useful)?

**Comment by D-side** [comment]
I'm on the fence about this, leaning towards "wait".

IMO it *is* useful for many purposes in its current form, but missing information on higher difficulties (as a result of missing JS) is a *significant* omission. I'm not sure what kinds of breakages are expected from the main library, but in this state I wouldn't announce its existence in Terraria communities (hypothetically, I'm not actively participating in any right now, this is solely to indicate my level of doubt). Is this enough of a bar to wait?

**Comment by GitEin11** [comment]
I think it is still good to make the current version available, their still a work around to access exclusive items thru the article  named "expert-exclusive content", although the stats like monster hp and such will still be missing

**Comment by Popolechien** [comment]
Ok let's wait a bit then.

**Comment by Markus-Rost** [comment]
mwoffliner now supports JS in the `main` branch (#1687), so I've been looking into the additional modules this ZIM will need:

- `site` (Made up of [MediaWiki:Common.js](https://terraria.wiki.gg/wiki/MediaWiki:Common.js) and [MediaWiki:Vector.js](https://terraria.wiki.gg/wiki/MediaWiki:Vector.js))
  - This is the module containing the Classic/Expert/Master mode tabs blocking the ZIM.
  - Sadly the module in it's current form is making some external requests and as such can't be trusted inside a ZIM.
    - **AJAX tables**: Used to load data from other pages on click.
    - **Hair Dyes slider**: Calling the API to parse some templates.
  - The module also has some dependencies on other JS modules. As the `site` module has no defined dependencies in the ResourceLoader, these will have to be defined manually as well.
    - `mediawiki.util`: Trusted module and all usages are safe.
    - `skins.vector.legacy.js`: Only modifies UI elements that aren't part of the ZIM, can be ignored (will cause a failed internal request).
    - `oojs-ui`: Only used for the `NumberInputWidget` by the Space altitude calculator, should be safe to include.
    - `jquery.tablesorter`: Trusted module, however only usage is inside the AJAX tables script.
- `ext.gadget.referenceTooltips` (Made up of [MediaWiki:Gadget-referenceTooltips.js](https://terraria.wiki.gg/wiki/MediaWiki:Gadget-referenceTooltips.js) and CSS)
  - Looks like the settings popup includes an external help icon, however the URL doesn't even resolve to a valid image ([line 478](https://terraria.wiki.gg/wiki/MediaWiki:Gadget-referenceTooltips.js#L-478)). Unsure if this minor instance is worth excluding the module. Only way to reach this is by manually opening the settings and disabling the tooltips, after which they also can't be enabled again because the location of the button is not included in the ZIM. @benoit74 what do you think? Wikipedia uses the same gadget so this question will come up there as well.
  - The module also has some conditional dependencies on other JS modules that aren't defined in the ResourceLoader, these will have to be defined manually as well.
    - `oojs`: All usages should be safe to include.
    - `oojs-ui`: All usages should be safe to include.
- `ext.gadget.flashAnchored` (Made up of [MediaWiki:Gadget-flashAnchored.js](https://terraria.wiki.gg/wiki/MediaWiki:Gadget-flashAnchored.js) and CSS)
  - Safe to include, no external requests.
- Other gadgets are either not enabled by default or not relevant inside a ZIM.
- `ext.SimpleMathJax`
  - Requests JS from a `wgExtensionAssetsPath` based file, which we can't scrape currently.
  - See also openzim/mwoffliner#1371

My suggestion would be for the wiki admins to move the two mentioned functions out of MediaWiki:Common.js, ideally into a new default enabled gadget. 

Once that has happened, the ZIM recipe should use `addModule=site,mediawiki.util,ext.gadget.flashAnchored,ext.gadget.referenceTooltips,oojs,oojs-ui`

**Comment by benoit74** [comment]
Regading `ext.gadget.referenceTooltips`, I feel like tooltip are a great feature, quite important for me to be inside the ZIM. Is it even possible to reach the settings popup from inside the ZIM (since users do not have access to their user settings anyway)?

**Comment by Markus-Rost** [comment]
The settings popup is working fine inside a ZIM as this is a cookie based setting.

**Comment by benoit74** [comment]
OK, got it. I feel like we should try to ask website admins to fix URL, and not care for not being able to reactivate tooltips once disabled, I consider "that's life".

Does anyone knows how to contact site admin to suggest implementing the extraction of `AJAX tables` and `Hair Dyes slider` into a gadget and fixing the URL used in `MediaWiki:Gadget-ReferenceTooltips.js` with WPEN one for instance: https://en.wikipedia.org/wiki/MediaWiki:Gadget-ReferenceTooltips.js ?

**Comment by schlegel42** [comment]
Is there any chance that someone could make a new .zim because of Terraria's 1.4.5 update? Or could someone tell me how I could do it myself? I don't care about the cosmetics or anything like that. I just desperately need an offline wiki for the new items, enemies, mechanics, and ask that other fun stuff.

**Comment by Popolechien** [comment]
I have requested the updated but my understanding is that some of the source issues have not been resolved. 
@schlegel42 Can you please contact the website admins and point them to [this comment](https://github.com/openzim/zim-requests/issues/735#issuecomment-4063414812) (see last line, about moving functions out of commons.js)?

**Comment by schlegel42** [comment]
The database is currently down for maintenance, so I can't create an account to put something on the administrative notice board. For anyone checking this out in the future, https://terraria.wiki.gg/wiki/Terraria_Wiki:Admin_noticeboard would be the place to go and add this request.

**Comment by Popolechien** [comment]
Website seems back up on my end. Page seems decently active.

**Comment by schlegel42** [comment]
I can confirm the database is back up and I was able to create an account. I posted the request, so now we just have to wait. I'm probably not the best person to continue interacting and checking up on this due to me not having internet 24/7. As such, I will recommend that someone else keeps up with this, but I'll try my best.

**Comment by Popolechien** [comment]
Thanks a lot.

**Comment by mQNwrK** [comment]
I came here from https://terraria.wiki.gg/wiki/Terraria_Wiki:Admin_noticeboard#Request_from_the_OpenZIM_community .

The parts blocking the `site` module are now offloaded to gadgets.
We do indeed simply import `ext.gadget.referenceTooltips` from the English Wikipedia; I updated it to the most recent version from there which fixes the broken URL.

Is there anything else that needs to be done on the wiki side before the ZIM file can be created? (I should note that our wiki host has been somewhat unstable in the last 6 months or so, so things like [this](https://terraria.wiki.gg/wiki/Terraria_Wiki:Admin_noticeboard#Various_expression,_lua_and_database_errors_when_not_logged_in) are not something we as editors can fix, unfortunately.)

**Comment by Popolechien** [comment]
Ran a successful run but maths is still missing, see e.g. https://staging.library.kiwix.org/viewer#terrariawiki_en_2026-07/Luck (impact section)

<img width="1052" height="393" alt="Image" src="https://github.com/user-attachments/assets/95ffda5e-ee31-4bde-870f-cb0d4adb31a5" />

**Comment by benoit74** [comment]
@Popolechien thank you for finding the page with MathJax problem, I was searching for such a page since quite long

I've reconfigured recipe to use latest (not yet released) mwoffliner version and to correctly add support for MathJax, additional JS modules which have been "fixed" and categories. Let's see but it might be a very promising ZIM to showcase what 2.0.0 is going to change.

@mQNwrK thank you very very much for your understanding and your very valuable help. Much appreciated for the community.

**Comment by benoit74** [comment]
We have a new ZIM: https://staging.library.kiwix.org/viewer#terraria_en_all_2026-07

It has fixed:
- categories are now present
- MathJax (equations) rendering 
- added JS modules (polishing some UI)

I've noticed margin and padding on outer content were missing so I've added a bit of custom CSS in Zimfarm configuration (for next ZIM), and I've also fixed recipe + ZIM names: https://farm.openzim.org/recipes/terraria.wiki.gg_en_all / https://cms.openzim.org/title/404bb8b8-f4de-491f-8f17-77599b277e7c

Waiting for your feedback before running a (last?) task to polish these (and anything else you might find broken)

**Comment by benoit74** [comment]
Edit: tabs on monster infobox like https://staging.library.kiwix.org/content/terraria_en_all_2026-07/Angry_Bones are still not working. We probably still miss a bit of JS.

**Comment by mQNwrK** [comment]
That's neat, thank you.

You're right, for the Classic/Expert/Master tabs there's still a JavaScript module missing: [`mediawiki.storage`](https://www.mediawiki.org/wiki/ResourceLoader/Core_modules#mediawiki.storage). It's used for remembering the reader's tab selection across pages. We added this functionality recently because some users found it cumbersome to keep having to click on e.g. the "Master" tab on every page when they're playing in that mode and aren't interested in the Classic Mode stats, which are shown on the wiki by default.

I don't know how feasible it would be to add this module? I'm not yet familiar with how it all works in the ZIM. If it's too much of a hassle, then I can add a fallback function where the tab-remembering functionality would be omitted but at least the tab switching would work.

I also noticed the warning "Loading failed for the <script> with source “[https://staging.library.kiwix.org/content/terraria_en_all_2026-07/_mw_/skins.vector.legacy.js.js](https://staging.library.kiwix.org/content/terraria_en_all_2026-07/_mw_/skins.vector.legacy.js.js)”." in the browser console; I suppose that's what MarkusRost already anticipated above. Is it necessary/desirable/possible to get rid of this warning? – no clue. Just thought I'd mention it.

**Comment by Markus-Rost** [comment]
`mediawiki.storage` seems to be safe and could even be added to the trusted js scripts in mwoffliner. There is still kiwix/overview#127, however mwoffliner should have this mostly mitigated by modifying the `vary` key in jsConfigVars with a timestamp. 

Leaving the `vector.skins.legacy.js` warning should be fine as is.

**Comment by mQNwrK** [comment]
I tweaked the wiki CSS a bit to improve the styling in the ZIM. For the custom ZIM CSS, I believe the following should suffice:

```css
body.skin-vector {
	display: block !important;
	padding: var(--layout-padding-y) var(--layout-padding-x) !important;

	@media (max-width: 720px) {
		padding: calc(var(--layout-padding-y) / 2) calc(var(--layout-padding-x) / 2) !important;
	}
}

#mw-content-text {
	min-height: unset !important;
}
```

I couldn't figure out where to edit this myself, so – @benoit74 Would you kindly replace `customCss` in the recipe config with the above?

Another thing: page titles show up as raw HTML, as can be seen in the browser tab title, browser window title, `document.title` on the console, etc. After a quick glance at mwoffliner, I gather that it asks the `action=parse` API for a page's displaytitle prop, but nothing stood out to me as to why it wouldn't parse the returned HTML. Here are some examples:

* https://staging.library.kiwix.org/content/terraria_en_all_2026-07/Weapons (regular mainspace page) → [API](https://terraria.wiki.gg/api.php?action=parse&prop=displaytitle&page=Weapons)
* https://staging.library.kiwix.org/content/terraria_en_all_2026-07/Legacy:Acorn (outside of mainspace) → [API](https://terraria.wiki.gg/api.php?action=parse&prop=displaytitle&page=Legacy:Acorn)
* https://staging.library.kiwix.org/content/terraria_en_all_2026-07/Terraria_2 (italic title, set in page source) → [API](https://terraria.wiki.gg/api.php?action=parse&prop=displaytitle&page=Terraria_2)
* https://staging.library.kiwix.org/content/terraria_en_all_2026-07/TModLoader (lower-cased first letter in title, set in page source) → [API](https://terraria.wiki.gg/api.php?action=parse&prop=displaytitle&page=TModLoader) (no HTML here, so this one works)
* https://staging.library.kiwix.org/content/terraria_en_all_2026-07/Rich_Mahogany_Sword/it (completely custom displaytitle, set in page source) → [API](https://terraria.wiki.gg/api.php?action=parse&prop=displaytitle&page=Rich_Mahogany_Sword/it)

**Comment by benoit74** [comment]
Thanks a lot @Markus-Rost and @mQNwrK ;

I've added the `mediawiki.storage` module (see [here](https://farm.openzim.org/recipes/terraria.wiki.gg_en_all/history?compare=bdf29de0-6bcd-4f6c-8ffc-817813224667...0e8483d4-75c5-49c5-991b-a4f3c083c99b)) , and restarted the recipe.

Regarding the custom CSS, how is this "superior" to the custom CSS I've already configured in https://farm.openzim.org/recipes/terraria.wiki.gg_en_all/history?compare=cf6ca327-6db9-4560-a480-7e3f5499af73...931b1d6c-9eaf-4471-b500-227702231e16 but not yet "applied" (like I did not restarted the recipe)?

Regarding page titles showing up as raw HTML, this is now fixed in `main` branch, was not limited to Terraria and should not show up in next ZIM (this is what happens when you use a cutting edge `dev` Docker image, you encounter totally unrelated bugs ^^).

**Comment by mQNwrK** [comment]
Great. Cool that you've got the displaytitles fixed already!

About the custom CSS – these two rules are crucial in my opinion for a proper layout:

```css
body.skin-vector {
	display: block !important;
}

#mw-content-text {
	min-height: unset !important;
}
```

The layout on the live site is designed for a bunch of interface elements around the actual content area, like the sidebar and navigation bar. It's done with a CSS grid (`body` has `display: grid`) and the content area `div.mw-body` is positioned in that grid. Since in the ZIM the content area is really the *only* child of `body`, the grid layout breaks and `div.mw-body` is mis-positioned. Therefore, it's easiest to just disable the (non-functional) grid altogether by changing `body { display: grid }` to `body { display: block }`.

Similarly, `#mw-content-text` has a `min-height` rule to make it look a bit more pleasant on short pages, adding some space above the categories box. In the ZIM, `div#catlinks` is *inside* of `div#mw-content-text`, though, so all the `min-height` rule does is add empty brown space at the very bottom of the page, which doesn't look good in my opinion. Thus, disabling it with `#mw-content-text { min-height: unset }`.

As I was browsing the ZIM I noticed that I forgot another rule: `body.skin-vector { min-height: unset !important; }`. It's necessary to remove a small amount of useless scrolling.

The padding rules are not as critical. I suppose you can keep them if you prefer. Except for this, though:

```css
#content.mw-body {
	@media (min-width: 768px) {
		margin-right: 0rem;
		margin-left: 0rem;
	}
}
```

I think it looks just awful when there's no space at all on either side.

Below I added some screenshots of the look of https://staging.library.kiwix.org/content/terraria_en_all_2026-07/Recipes/Teapot (an exceptionally short page, but it's good for demonstration) on two different viewport sizes (~450px width and ~1920px width), to show what I mean with the layout fixes.

----

With this `customCss` (currently configured in the recipe):

```css
#content.mw-body {
	margin-top: 2rem;
	margin-bottom: 2rem;

	@media (min-width: 768px) {
		padding: 0.5rem;
		margin-right: 0rem;
		margin-left: 0rem;
	}

	@media (min-width: 992px) {
		padding: 0.5rem;
		margin-right: 1rem;
		margin-left: 1rem;
	}

	@media (min-width: 1200px) {
		padding: 1rem;
		margin-right: 2rem;
		margin-left: 2rem;
	}
}
```

<details>
<summary>Screenshots</summary>

<img width="241" height="520" alt="Small-screen screenshot of the page Recipes/Teapot" src="https://github.com/user-attachments/assets/14d1a9f2-5033-469d-a522-506fc24fb918" />

<img width="960" height="540" alt="Wide-screen screenshot of the page Recipes/Teapot" src="https://github.com/user-attachments/assets/c5c05af8-1182-4c04-b211-612865ac4d60" />

</details>

----

With this `customCss`:

```css
body.skin-vector {
	display: block !important;
	min-height: unset !important;
	padding: var(--layout-padding-y) var(--layout-padding-x) !important;

	@media (max-width: 720px) {
		padding: calc(var(--layout-padding-y) / 2) calc(var(--layout-padding-x) / 2) !important;
	}
}

#mw-content-text {
	min-height: unset !important;
}
```

<details>
<summary>Screenshots</summary>

<img width="241" height="520" alt="Small-screen screenshot of the page Recipes/Teapot" src="https://github.com/user-attachments/assets/e69e9218-3a20-48e2-9210-bdd1605b6e9e" />

<img width="960" height="540" alt="Wide-screen screenshot of the page Recipes/Teapot" src="https://github.com/user-attachments/assets/bf11e895-3c74-4c14-a8d8-236de7714f36" />

</details>

----

Keep in mind that I changed the wiki CSS since your creation of the ZIM: https://terraria.wiki.gg/wiki/MediaWiki:Common.css?diff=1003629&oldid=1000380 . These changes are already incorporated in all four screenshots above.

**Comment by benoit74** [comment]
Thanks a lot for all these details.

Note that recipe is currently failing, upstream issue to follow is https://github.com/openzim/mwoffliner/issues/2836

I've updated the CSS in recipe, waiting for next run once upstream issue is solved.

**Comment by benoit74** [comment]
And we have a new ZIM: https://staging.library.kiwix.org/viewer#terraria.wiki.gg_en_all_2026-07 

I had a look and I feel like we have solved all known issues. 

@Popolechien @mQNwrK @D-side please review this new ZIM

**Comment by Popolechien** [comment]
Had a quick pass and found no issue \o/

**Comment by GitEin11** [comment]
while the online version works when switching casual to master/expert mode, in android kiwix version it doesn't switch. I guess its a bug with the app itself and not with the zim.

update: I downloaded the wrong zim file hahaha. there are 2 zim with 1.19 Gb and 1.14 Gb. the 1.14 works without issue.

**Comment by D-side** [comment]
Math on the [Luck page](https://staging.library.kiwix.org/viewer#terraria.wiki.gg_en_all_2026-07/Luck) is all wrapped in containers with `style="opacity:.5"`, which isn't the case [in the origin](https://terraria.wiki.gg/wiki/Luck#Impact) (it's `style="opacity: 1"` instead). Contents seem to have rendered properly though, which is a major improvement. And it's semi-readable (pun very much intended) even as it is.

**Comment by Popolechien** [comment]
Moving to PROD.

**Comment by benoit74** [comment]
@mQNwrK any chance we can fix what causes the opacity to be 0.5 instead of 1? Thank you very much for your help crafting this ZIM anyway. We should now proceed with other languages asap.

**Comment by mQNwrK** [comment]
Thanks for the new ZIM, looking great already.

There's a function in the SimpleMathJax extension that sets the opacity of the math elements to 1 after the math is parsed (I think?):

https://github.com/jmnote/SimpleMathJax/blob/main/resources/ext.SimpleMathJax.js#L130

```javascript
pageReady: () => {
    return MathJax.startup.defaultPageReady().then(() => {
        $("span.smj-container > .MathJax").parent().css('opacity',1);
    });
}
```

As far as I understand it, the MathJax config part is included in the ZIM, but this hook is probably missing, right? Unfortunately I couldn't say if or how it can be added, since my understanding of all the MathJax-related stuff is rather superficial. At the moment the function does nothing else apart from setting the opacity, so I could also set a forced `.smj-container { opacity: 1 !important }` in the wiki CSS. Although that would mean that the "un-parsed" math text (that shows up briefly on page load, e.g. `[math]\displaystyle{ \ointctrclockwise E=mc^2 }[/math]` on https://terraria.wiki.gg/wiki/User:KiwixOffline/MWOffliner_Test ) would also get opacity:1. Let me know if I should do that or if there's a way to get that function into the ZIM.

Something else: It seems the animated PNGs are still missing. For example, compare https://staging.library.kiwix.org/content/terraria.wiki.gg_en_all_2026-07/Cages#Types or https://staging.library.kiwix.org/content/terraria.wiki.gg_en_all_2026-07/Butterfly_jars#Types or https://staging.library.kiwix.org/content/terraria.wiki.gg_en_all_2026-07/Dragonfly_jars#Types with the respective live articles – there should be images in the "Placed" column. (It's mostly just these three articles and a couple others that use animated PNGs, so not a *huge* problem.) Should I mention this in the mwoffliner issue?

----

Edit to add: MP4 videos are also not included (e.g. https://staging.library.kiwix.org/content/terraria.wiki.gg_en_all_2026-07/Biome_spread , top right corner), which is not intentional, is it? This is from the scraper log:

```
[2026-07-21 10:01:39.496] WARN: Deleting audio/video HTML node missing an appropriate source in page at 'Biome_spread':
<video poster="/images/thumb/Biome_spread_%28demo%29.mp4/320px--Biome_spread_%28demo%29.mp4.jpg?366eb6" controls="40" preload="none" class="thumbimage" width="320" height="310" data-durationhint="13" data-mwtitle="Biome_spread_(demo).mp4" data-mwprovider="local" style="max-width:100%;max-height:100%;height:100%"></video>
```

**Comment by benoit74** [comment]
Thanks a lot for the pointer to the SimpleMathJax extension, this is indeed the issue. Not really sure how to hook this either, but we can probably find at least a semi-working fix, I will dig into it later.

I will also dig into the PNG issue, and reopen issue on mwoffliner side but I tested on the `Cages` so it is not exactly the problem I already solved. Maybe our image cache is corrupted for these animated images which did not worked correctly in the past ... I will dig into this.

And regarding MP4, this is a known scraper issue and a matter of debate, see https://github.com/openzim/mwoffliner/issues/2745 and https://github.com/openzim/mwoffliner/pull/2747 (basically I consider we should prefer webm and default to any video format available ; Kelson want only webm in the ZIM for broader compatibility ... but this is rather more complex to implement and I'm probably too lazy)

**Comment by benoit74** [comment]
I failed to make the startup function work as expected, looks like it does not find jQuery ... so I've simply overridden the CSS rule to set opacity to 1 from page load (which removes the fade-in effect once MathJax is loaded, but I suppose this is acceptable and better than staying at 50% opacity forever).

I've hopefully fixed issue in scraper regarding missing images, it was indeed a problem of corrupted S3 cache.

I've just restarted the Zimfarm recipe to create a new ZIM.

**Comment by Popolechien** [comment]
- [x] Available at https://browse.library.kiwix.org/#lang=eng&q=terraria
- [x] Announced on Mastodon
- [x] Added to Newsletter list

**Comment by mQNwrK** [comment]
Wonderful! Thank you all very much for your hard work.

What's the plan regarding ZIM updates going forward? Would it be possible to have scheduled periodic updates?

**Comment by D-side** [comment]
@mQNwrK [yep](https://github.com/openzim/zim-requests/issues/735#issuecomment-3471536249):

> We aim to update the ZIM quarterly. It is fully automated. It allows to reasonably grab any edits made in the Wiki, and to benefit from any new feature / bug fix done in the scraper.
> 
> You can always request (with a new issue in this Github Repo) a manual run between these quarterly updates if for some reason the ZIM is too outdated because many edits have been done in the wiki (typically following a major release). But we prefer when we can avoid them to save our efforts for other stuff ^^

**Comment by mQNwrK** [comment]
Ah, right, thanks. Does this need to be configured in the Zimfarm? I just checked and https://farm.openzim.org/recipes/terraria.wiki.gg_en_all says "Periodicity: manually", or does that refer to something else?

**Comment by D-side** [comment]
I don't *know*, but I think you're right, [Stardew Valley's recipe](https://farm.openzim.org/recipes/stardewvalley_wiki) says `biannually` there.
So that quoted comment was probably a plan for the future, which is… probably right about now.

**Comment by mQNwrK** [comment]
Alright, makes sense. @benoit74 Would you mind enabling the quarterly ZIM updates, please?

Also, this is a very minor thing, but while writing a wiki help page about this, I noticed that the link to the Kiwix online viewer only works with the timestamp and not the "permalink":

- https://browse.library.kiwix.org/viewer#stardewvalleywiki_en_all and https://browse.library.kiwix.org/viewer#stardewvalleywiki_en_all_2026-07 both work,
- but https://browse.library.kiwix.org/viewer#terraria.wiki.gg_en_all is not found, only https://browse.library.kiwix.org/viewer#terraria.wiki.gg_en_all_2026-07a .

Again, just a very minor inconvenience and not a problem if not fixed. But maybe it's a hint that something else is amiss.

**Comment by Popolechien** [comment]
The default periodicity is indeed quarterly, which is now set for both Terraria and Stardew Valley.
I suspect that https://browse.library.kiwix.org/viewer#terraria.wiki.gg_en_all pointed to https://browse.library.kiwix.org/viewer#terraria.wiki.gg_en_all_2026-07 (which has been deleted) and has not been reassigned to 2026-07a, but that's just a guess.

**Comment by benoit74** [comment]
I've opened https://github.com/kiwix/libkiwix/issues/1318 to fix the permalink issue, thank you very much for reporting, this is indeed unexpected.
